wrappers/opencv/depth-filter/rs-depth-filter.cpp

```diff
@@ -0,0 +1,310 @@
+        cv::addWeighted(area->abs_scharr_y, 0.5, area->abs_scharr_y, 0.5, 0, area->edge_mask);
+        cv::threshold(area->edge_mask, area->edge_mask, 192, 255, cv::THRESH_BINARY);
+    }
+
+    void init_matrices(const cv::Mat& ir_resized) 
+    {
+        const int sizeX = ir_resized.cols;
+        const int sizeY = ir_resized.rows;
+        const int sizeYperSubImage = sizeY / NUM_SUB_IMAGES;
+        const int sizeXtimesSizeY = sizeX * sizeY;
+
+        // test on one case if something has changed - this would then apply to all other image matrices as well
+        bool needToReinitialize = _mask_edge.size() != ir_resized.size() || _mask_edge.type() != CV_8U;
+
+        if (!needToReinitialize)
+        {
+            // this needs resetting every frame
+            memset(_depth_output.data, 0, 2 * sizeXtimesSizeY);
+            return;
+        }
+
+        static_assert(NUM_SUB_IMAGES >= 1 && NUM_SUB_IMAGES < 64, "NUM_SUB_IMAGES value might be wrong");
+
+        _mask_edge      = cv::Mat(ir_resized.size(), CV_8U);
+        _mask_harris    = cv::Mat(ir_resized.size(), CV_8U);
+        _mask_combined  = cv::Mat(ir_resized.size(), CV_8U);
+        _depth_output   = cv::Mat(ir_resized.size(), CV_16U);
+        _ir_float       = cv::Mat(ir_resized.size(), CV_32F);
+        _corners        = cv::Mat(ir_resized.size(), CV_32F);
+        _scharr_x       = cv::Mat(ir_resized.size(), CV_16S);
+        _scharr_y       = cv::Mat(ir_resized.size(), CV_16S);
+        _abs_scharr_x   = cv::Mat(ir_resized.size(), CV_8U);
+        _abs_scharr_y   = cv::Mat(ir_resized.size(), CV_8U);
+
+        // this needs to be cleared out, as we only copy to it with a mask later
+        memset(_depth_output.data, 0, 2 * sizeXtimesSizeY);
+
+        std::vector<cv::Rect> rects;
+        for (int i = 0; i < NUM_SUB_IMAGES; i++) 
+        {
+            cv::Rect rect(0, i * sizeYperSubImage, sizeX, sizeYperSubImage);
+            rects.push_back(rect);
+        }
+
+        for (auto&& rect : rects)
+        {
+            sub_area si;
+            si.decimated_depth   = _decimated_depth(rect);
+            si.decimated_ir      = _decimated_ir(rect);
+            si.edge_mask         = _mask_edge(rect);
+            si.harris_mask       = _mask_harris(rect);
+            si.combined_mask     = _mask_combined(rect);
+            si.output_depth      = _depth_output(rect);
+            si.float_ir          = _ir_float(rect);
+            si.corners           = _corners(rect);
+            si.scharr_x          = _scharr_x(rect);
+            si.scharr_y          = _scharr_y(rect);
+            si.abs_scharr_x      = _abs_scharr_x(rect);
+            si.abs_scharr_y      = _abs_scharr_y(rect);
+            sub_areas.push_back(si);
+        }
+    }
+};
+
+int main(int argc, char * argv[]) try
+{
+    rs2::colorizer color_map;
+
+    rs2::pipeline pipe;
+
+    rs2::config cfg;
+    cfg.enable_stream(RS2_STREAM_DEPTH, 848, 480);
```

Comment:
The params are D435-oriented. Can the explicit params be replaced with defaults to allow using D415 and SR300?